### PR TITLE
Added OrbitPlotter2D Class for plotting in 2D using Plotly

### PR DESCRIPTION
**Pull request by shreyasbapat**
Earlier we didn't have an option to plot objects in 2D using plotly. 
screenshot from 2018-04-04 00-46-58
This solves #338

### CODE REVIEW

**Comment by shreyasbapat** [comment]
@Juanlu001  The same test was present in ``OrbitPlotter3D`` but there, the cyclomatic complexity was fine. Any reason why it is causing problems?

**Comment by Juanlu001** [comment]
The problem with cyclomatic complexity in tests is that `assert` statements count as a branch, but it's a bit debatable:

https://github.com/rubik/radon/issues/42

Please bring back the assertions that you removed and I will manually approve the issue in Codeclimate.

**Comment by Juanlu001** [comment]
Can you please remove this one @anhiga? :smile:

**Comment by shreyasbapat** [comment]
No. I didn't remove any of the assertions. Those were actually not needed. (Z axis) :see_no_evil:

**Comment by shreyasbapat** [comment]
@Juanlu001  What can be the next target?

**Comment by Juanlu001** [comment]
@shreyasbapat I have my own opinion but I want to hear @anhiga first :)

**Comment by anhiga** [comment]
Hi @shreyasbapat, sorry for my late introduction. It is nice to see you making such improvements in `poliastro`.
This PR looks really great, and it looks like you are early addressing and completing some of the tasks of your proposal, which is also nice 😄. However I have lately been away from poliastro code (much more time than I shoud), so I will try to catch up with all the changes, issues and PRs submitted, and make a deep review of this particular Pull Request as soon as possible.

At first sight, I would try to add as many function docstrings as possible (I am aware that many functions before this PR were also lacking docstrings, but that doesn't make them less important 😆 ).

**Comment by shreyasbapat** [comment]
Hi @anhiga ! I will do two things before you review : 
1) Add more test cases
2) Add docstrings to functions.
Thanks! :)

File: src/poliastro/plotting.py [hunk 1]
```diff
@@ -394,6 +394,226 @@ def savefig(self, filename, **layout_kwargs):
...
+    yy = y_center + radius * np.sin(u1)
+
+    return xx, yy
+
+
+class OrbitPlotter2D:
+    """OrbitPlotter2D class.
+
+    """
+
```
Comment (Juanlu001) [inline src/poliastro/plotting.py]: Extra whitespace?

File: src/poliastro/plotting.py [hunk 2]
```diff
@@ -394,6 +394,226 @@ def savefig(self, filename, **layout_kwargs):
...
+        """
+        if self._attractor is None:
+            self._attractor = attractor
+
+        elif attractor is not self._attractor:
+            raise NotImplementedError("Attractor has already been set to {}.".format(self._attractor.name))
+
+    @u.quantity_input(elev=u.rad, azim=u.rad, distance=u.km)
+    def set_view(self, elev, azim, distance=5 * u.km):
```
Comment (Juanlu001) [inline src/poliastro/plotting.py]: Does this make sense for 2D plots?

File: src/poliastro/tests/test_plotting2d.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    assert figure["data"] == [{}]
+    assert figure["layout"]["autosize"] is True
+    assert "xaxis" in figure["layout"]["scene"]
+    assert "yaxis" in figure["layout"]["scene"]
+    assert "aspectmode" in figure["layout"]["scene"]
+
+
+def test_set_different_attractor_raises_error():
+    body1 = mock.MagicMock()
```
Comment (Juanlu001) [inline src/poliastro/tests/test_plotting2d.py]: Do we need Mocks here? Can't we use the `Earth` and `Mars` for instance?

File: src/poliastro/tests/test_plotting2d.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    frame.set_attractor(body1)
+
+    with pytest.raises(NotImplementedError) as excinfo:
+        frame.set_attractor(body2)
+    assert "Attractor has already been set to 1." in excinfo.exconly()
+
+
+def test_plot_sets_attractor():
+    frame = OrbitPlotter2D()
+    assert frame._attractor is None
```
Comment (Juanlu001) [inline src/poliastro/tests/test_plotting2d.py]: Not sure whether we should test against private properties, but on the other hand I don't have a better idea :confused:

File: src/poliastro/plotting.py (hunk 1 above)
Comment (shreyasbapat) [reply src/poliastro/plotting.py]: Oh! Will remove that!

File: src/poliastro/plotting.py (hunk 2 above)
Comment (shreyasbapat) [reply src/poliastro/plotting.py]: No! But as we plan to further merge the 3D and 2D classes, so I kept it. I will remove it!

File: src/poliastro/tests/test_plotting2d.py (hunk 3 above)
Comment (shreyasbapat) [reply src/poliastro/tests/test_plotting2d.py]: Noted!

**Comment by shreyasbapat** [comment]
I was a bit busy these days! I will make some more tests today itself!

File: src/poliastro/plotting.py (hunk 1 above)
Comment (shreyasbapat) [reply src/poliastro/plotting.py]: I followed the way ``OrbitPlotter3D`` is written

Comment (shreyasbapat) [reply src/poliastro/plotting.py]: https://github.com/shreyasbapat/poliastro/blob/5da5388dd111df0a9701071530e1a2c5920ca21c/src/poliastro/plotting.py#L236

File: src/poliastro/tests/test_plotting2d.py (hunk 4 above)
Comment (shreyasbapat) [reply src/poliastro/tests/test_plotting2d.py]: What is your suggestion?

**Comment by shreyasbapat** [comment]
Rebased after #341

**Comment by shreyasbapat** [comment]
@Juanlu001  W504 showing up in many files. @nikita-astronaut  did your PR cause this? Should I fix this?

**Comment by Juanlu001** [comment]
@shreyasbapat re: W504 no, @nikita-astronaut did not cause it. It's because an upgrade in pycodestyle. Hold on to fix it, in case we fix the other one faster.

File: src/poliastro/tests/test_plotting2d.py (hunk 4 above)
Comment (Juanlu001) [reply src/poliastro/tests/test_plotting2d.py]: Don't have a better suggestion really, just leave it like it is.

Comment (shreyasbapat) [reply src/poliastro/tests/test_plotting2d.py]: Okay!

**Comment by Juanlu001** [comment]
@shreyasbapat can you rebase this? There are relevant changes in Codeclimate and pycodestyle in `master` now.

**Comment by shreyasbapat** [comment]
Rebased!

**Comment by Juanlu001** [comment]
Don't worry about Codeclimate, I will take care of that one.

**Comment by shreyasbapat** [comment]
@Juanlu001  @anhiga  Any other comments?

**Comment by shreyasbapat** [comment]
Maybe we can also include this in 0.9!?

**Comment by Juanlu001** [comment]
This looks good to me, will get rid of the codeclimate warning and merge
this weekend

**Comment by Juanlu001** [comment]
Merging!

**Comment by shreyasbapat** [comment]
Thanks! :)

**Pull request merged**